app/models/course_module.rb

```diff
@@ -2,10 +2,14 @@
 
 class CourseModule < ApplicationRecord
   belongs_to :course_year
-  has_one :next_module, class_name: "CourseModule", foreign_key: :next_module_id
-  belongs_to :previous_module, class_name: "CourseModule", inverse_of: :next_module, optional: true
   has_many :course_lessons, dependent: :delete_all
 
+  # We use previous_module_id to store the connections between lessons
```

Comment:
Should that be connections between modules.

Reply:
I just wonder if we should perhaps update the schema rather than adding the comments. Is this more a result of a bad schema.

Reply:
> Should we replace lesson with module in the text.
Good spot!

> I just wonder if we should perhaps update the schema rather than adding the comments. Is this more a result of a bad schema.
I am happy to make a better schema, just don't know how. The little I understand about Rails tells me there isn't a pretty solution.

I want a chain of lessons. I think pointing to the previous lesson from current one makes perfect sense to achieve that. Hence `belongs_to` previous lesson, cause id lives on current lesson. (at least I think that's what Rails expects - id lives on our object = our object belongs to some other object)

Then if we want to access next lesson, it needs to be via `has_one`, cause we don't store that id on our current lesson. And if it has to be `has_one`, then it needs a field from the target object to tell it what is the name, on the target objet, to look for. And `previous_lesson_id` makes sense as well.

I don't think it's a bad schema, because it seems that the `next_lesson` requiring `previous_lesson_id` as foreign key is a perfectly natural result of Rails philosophy coupled with reasonable database structure. I put a comment in because it is counter-intuitive to me. And I don't know how to make it intuitive in Rails. 

We could, I guess, duplicate the connections - have both previous lesson and next lesson ids on the model - but then we risk them getting out of sync. And we store the same relation twice.